ako-gateway-api/k8s/gateway_controller.go

```diff
@@ -499,13 +501,20 @@ func (c *GatewayController) SetupGatewayApiEventHandlers(numWorkers uint32) {
 				utils.AviLog.Debugf("key: %s, msg: same resource version returning", key)
 				return
 			}
-			if !IsValidGateway(key, gw) {
+			valid, allowedRoutesAll, gwWithStatus := IsValidGateway(key, gw)
+			if !valid {
 				return
 			}
+			listRoutes, _ := validateReferreedHTTPRoute(key, gwWithStatus, allowedRoutesAll)
```

Comment:
Should we handle error returned by the validateRefereedHTTPRoute function ?

Reply:
Have logged error in that case.